### DISCUSSION TITLE
Change default teardown_module() name.

### DISCUSSION THREAD

**Discussion by EugenSusurrus** [pytest-dev/pytest#10557] [General]
Is there a way to change the `setup_module` function name? I am implementing a custom plugin which runs custom test cases, the test cases are formatted like this:

```python

def setUp():
    ...

def step_1():
    ...

def step_2():
    ...

def main():
    step_1()
    step_2()

def tearDown():
    ...
```

I can configure the pytest collection to collect test cases which match `main` like as stated in the documentation.

```
[pytest]
python_functions = main
```

My question is, can I change the plugin somehow to also collect `tearDown()` instead of the default `teardown_module(module)`
as stated in the documentation?

https://docs.pytest.org/en/7.1.x/example/pythoncollection.html#changing-naming-conventions

**Comment by nicoddemus** [comment]
Hi,

Currently no, that name is hardcoded. I think however you can implement a plugin which does that.

**Comment by EugenSusurrus** [comment]
@nicoddemus Wow amazingly fast reply!

I have some experience with writing pytest plugins already and feel quite comfortable with that.

Would you have a minimal example which you could share as a starting point? Or give me an idea as to what hook function would be best to use?

I was considering to somehow change the function name from `tearDown()` to `teardown_module()` when the module is imported in the `pytest_pycollect_makemodule()` hook. But this feels quite unnatural and I'm not sure if this is the right way.

**Comment by EugenSusurrus** [comment]
I might have found something which could help:

https://docs.pytest.org/en/7.1.x/example/special.html#a-session-fixture-which-can-look-at-all-collected-tests
